### PR TITLE
msggen: Enable Decode and DecodePay RPC

### PR DESCRIPTION
**Pull request by 2Fast2BCn**

### CODE REVIEW

**Comment by 2Fast2BCn** [comment]
Found a problem with my PR. Somehow `description` is lost.

**Comment by 2Fast2BCn** [comment]
Finally managed to get an actual result:

image

File: doc/schemas/listclosedchannels.schema.json [hunk 1]
```diff
@@ -98,11 +98,11 @@
...
-                    "static_remotekey/even",
-                    "anchor_outputs/even",
-                    "anchors_zero_fee_htlc_tx/even",
-                    "scid_alias/even",
-                    "zeroconf/even"
+                    "static_remotekey_even",
+                    "anchor_outputs_even",
+                    "anchors_zero_fee_htlc_tx_even",
+                    "scid_alias_even",
+                    "zeroconf_even"
```
Comment (rustyrussell) [inline doc/schemas/listclosedchannels.schema.json:121]: Yeah, you can't do this, these are the actual strings we output.  An "enum" in the schema just means a subset of possible values: if the / is bad for Rust, it needs to change it there...

File: doc/lightning-listpeerchannels.7.md [hunk 2]
```diff
@@ -194,4 +194,7 @@ Main web site: <https://github.com/ElementsProject/lightning> Lightning
 RFC site (BOLT \#9):
 <https://github.com/lightningnetwork/lightning-rfc/blob/master/09-features.md>
 
-[comment]: # ( SHA256STAMP:269c1f688cb6b3d42da9899386eabc48691a78987cfb455fafc4c2656df2b7df)
+<<<<<<< HEAD
+=======
+>>>>>>> 9d1d65a1b (msggen: Regenerate for ListPeerChannels)
+[comment]: # ( SHA256STAMP:c8cc5cce6119eeec5551d513f7cf60fdd21842f0f446263a37cce3fa3d0c1e33)
```
Comment (rustyrussell) [inline doc/lightning-listpeerchannels.7.md]: Oops, you committed a conflict marker!

Hmm, perhaps we should propagate "added" flags downwards in msggen?  By adding it to every field, it's almost useless for the documentation.  Though in this case, adding it to the schema fields is probably overkill anyway?

File: doc/schemas/decodepay.request.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    "bolt11"
+  ],
+  "properties": {
+    "bolt11": {
+      "type": "string",
+      "added": "v23.05"
+    },
+    "description": {
+      "type": "string",
+      "added": "v23.05"
```
Comment (rustyrussell) [inline doc/schemas/decodepay.request.json:16]: These added fields are unnecessary: while these are being added to GRPC now, they existed previously.

File: doc/schemas/decodepay.schema.json [hunk 4]
```diff
@@ -1,6 +1,7 @@
 {
   "$schema": "http://json-schema.org/draft-07/schema#",
   "type": "object",
+  "added": "v23.05",
```
Comment (rustyrussell) [inline doc/schemas/decodepay.schema.json:4]: These too: it really messes up our documentation!

File: doc/schemas/listclosedchannels.schema.json [hunk 5]
```diff
@@ -9,8 +9,10 @@
   "properties": {
     "closedchannels": {
       "type": "array",
+      "added": "v23.05",
```
Comment (rustyrussell) [inline doc/schemas/listclosedchannels.schema.json:12]: These too...

**Review by rustyrussell — CHANGES_REQUESTED** [review]
I'm going to leave the bulk of this for @cdecker to review, but I've left some comments.

It will also need Changelog-Added lines where grpc has been expanded...

File: doc/schemas/listclosedchannels.schema.json (hunk 5 above)
Comment (2Fast2BCn) [reply doc/schemas/listclosedchannels.schema.json:12]: I suspected as much. I'll remove them and update the PR.

Comment (2Fast2BCn) [reply doc/schemas/listclosedchannels.schema.json]: I removed `"added": "v23.05",` from all the schema json files (where I introduced them).
If you want me to squash my part of this PR then let me know.

File: doc/schemas/listclosedchannels.schema.json (hunk 1 above)
Comment (2Fast2BCn) [reply doc/schemas/listclosedchannels.schema.json]: I restored them back to their original state.
I'll wait for the feedback if this should be converted into an enum in primitives (like in my initial PR that I closed in favor of this one) or stay as string.

**Pull request closed without merging**